python-squid/src/main/java/org/sonar/python/tree/PythonTreeMaker.java

```diff
@@ -1020,8 +1026,27 @@ private PyArgListTree argList(AstNode argList) {
     return null;
   }
 
+  /*
+   * Post Condition on Generator Expression: parentheses can be omitted on calls with only one argument.
+   * https://docs.python.org/3/reference/expressions.html#grammar-token-generator-expression
+   */
+  private static void checkGeneratorExpressionInArgument(List<PyArgumentTree> arguments) {
+    List<PyArgumentTree> nonParenthesizedGeneratorExpressions = arguments.stream()
+      .filter(arg -> arg.expression().is(Tree.Kind.GENERATOR_EXPR) && !arg.expression().firstToken().getValue().equals("("))
+      .collect(Collectors.toList());
+    if (!nonParenthesizedGeneratorExpressions.isEmpty() && arguments.size() > 1) {
+      throw new RecognitionException(nonParenthesizedGeneratorExpressions.get(0).firstToken().getLine(), "Generator expression must be parenthesized.");
```

Comment:
The message should contain the line number so that it appears in the logs of the analysis.